app/controllers/api_controller/parser.rb

```diff
@@ -166,6 +174,22 @@ def parse_fetch_tenant(data)
       end
     end
 
+    def parse_fetch_provider(data)
+      if data
```

Comment:
Similarly, could we move this uncertainty over `data` up into the caller of this method?